src/ecuacion/MayorMenorNotas.java

```diff
@@ -0,0 +1,181 @@
+package ecuacion;
+import java.util.Scanner;
+import java.time.LocalDate;
+public class MayorMenorNotas {
+
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		boolean program = false;
+		int menosNotas= Integer.MAX_VALUE;
+		int masNotas = Integer.MIN_VALUE;
+		int menorPromedio = Integer.MAX_VALUE;
+		int mayorPromedio = Integer.MIN_VALUE;
+		String menosCantidadNotas="";
+		String masCantidadNotas="";
+		String mejorPromedio="";
+		String peorPromedio="";
+		int aux = 0;
+		int mayorEdad=0;
+		int menorEdad=0;
+		int masculino=0;
+		int femenino=0;
+		int programacion = 0;
+		int fisica=0;
+		int matematica=0;
+		int materia=0;
+		do {	
+			System.out.println("Cuantos estudiantes desea ingresar?");
+			int numStudent = scan.nextInt();
+			for(int i=0; i<numStudent;i++) {
+				System.out.println("Nombre del estudiante numero "+ (aux+1));
+				scan.nextLine();
+				String nameStudent = scan.nextLine();
+				//Fecha de nacimiendo del estudiante
+				System.out.println("Ingrese la fecha de nacimiento del estudiante del estudiante: ");
+				System.out.print("Ingrese el año: ");
+				int yearBorn= scan.nextInt();
+				System.out.print("Ingrese el mes: ");
+				int monthBorn= scan.nextInt();
+				System.out.print("Ingrese el dia: ");
+				int dayBorn= scan.nextInt();
+				
+				//Sexo del estudiante
+				System.out.println("Ingrese el tipo sexo del estudiante H/Hombre  M/Mujer");
+				String sex = scan.next();
+				switch(sex) {
```

Comment:
te recomiendo no usar mucho el switch, es considerado un antipatron en la propia programación, despues lo entenderas, ahh y otra cosa.

en Java, cuando comparamos 2 strings NUNCA se usa el operador ==

SIEMPRE se debe usar el metodo .equals de la clase string

algo asi 

```
if(sex.equals("H")){
  // es hombre
}
```

incluso si quieres comparar una letra y que haga match siendo mayuscula o minuscula puedes usar .equalsIgnoreCase("string a comparar")

algo asi:

```
if(sex.equalsIgnoreCase("H")){
  // es hombre aunque haya ingresado h minuscula. 
}
```

esto es porque el == solo se usa para variables primitivas y los strings en Java son objetos de la clase String.

esto se debe a que si comparas un objeto con == no estas comparando su valor, sino su referencia en memoria. 

**Investigar que es memoria heap y memoria stack en Java.**

Reply:
Verga gracias con esto, así puedo mejorar mucho más, todas estas correcciones se van a borrar?

Reply:
Como así?, ósea el PR queda ahí y todos los comentarios quedan, mientras no se borre el pr queda ahí y solo tienes que filtrar por PRs cerrados, si no lo que veas lo anotas en un block de notas en la pc o en tu chat de WhatsApp.

Reply:
redy to push